src-tauri/src/lib.rs

```diff
@@ -193,7 +215,42 @@ async fn do_install(nmod: &thermite::model::ModVersion, game_path: &std::path::P
     let download_path = format!("{}/{}", download_directory, filename);
     log::info!("Download path: {download_path}");
 
-    let nfile = thermite::core::manage::download_file(&nmod.url, download_path).unwrap();
+    let last_emit = RefCell::new(Instant::now()); // Keep track of the last time a signal was emitted
+    let nfile = thermite::core::manage::download_file_with_progress(
+        &nmod.url,
+        download_path,
+        |delta, current, total| {
+            if delta != 0 {
+                // Only emit a signal once every 250ms
+                let time_since_last_emit = Instant::now().duration_since(*last_emit.borrow());
+                if time_since_last_emit >= Duration::from_millis(250) {
+                    window
+                        .emit(
+                            "northstar-install-download-progress",
+                            InstallProgress {
+                                current_downloaded: current,
+                                total_size: total,
+                                state: InstallState::DOWNLOADING,
+                            },
+                        )
+                        .unwrap();
+                    *last_emit.borrow_mut() = Instant::now();
+                }
+            }
+        },
+    )
+    .unwrap();
+
+    window
```

Comment:
I know but I have to figure out how to get the information in the backend first which is gonna take me a bit.

Given that extraction time is rather static compared to download time I focused on finishing up what I already have. Probably best to add that in a separate PR so that we can roll out download progress bar ASAP ^^